CMakeLists.txt

```diff
@@ -92,31 +93,103 @@ option(IREE_HAL_DRIVER_VMVX "Enables the 'vmvx' runtime HAL driver" ${IREE_HAL_D
 option(IREE_HAL_DRIVER_VMVX_SYNC "Enables the 'vmvx-sync' runtime HAL driver" ${IREE_HAL_DRIVER_DEFAULTS})
 option(IREE_HAL_DRIVER_VULKAN "Enables the 'vulkan' runtime HAL driver" ${IREE_HAL_DRIVER_VULKAN_DEFAULT})
 
-cmake_dependent_option(IREE_TARGET_BACKEND_CUDA "Enables the 'cuda' compiler target backend" OFF ${IREE_BUILD_COMPILER} OFF)
-cmake_dependent_option(IREE_TARGET_BACKEND_DYLIB_LLVM_AOT "Enables the 'dylib-llvm-aot' compiler target backend" ON ${IREE_BUILD_COMPILER} OFF)
-cmake_dependent_option(IREE_TARGET_BACKEND_METAL_SPIRV "Enables the 'metal-spirv' compiler target backend" ON ${IREE_BUILD_COMPILER} OFF)
-cmake_dependent_option(IREE_TARGET_BACKEND_ROCM "Enables the 'rocm' compiler target backend" ON ${IREE_BUILD_COMPILER} OFF)
+message(STATUS "IREE runtime drivers:")
+if(IREE_HAL_DRIVER_CUDA)
+  message(STATUS "  - cuda")
+endif()
+if(IREE_HAL_DRIVER_DYLIB)
+  message(STATUS "  - dylib")
+endif()
+if(IREE_HAL_DRIVER_DYLIB_SYNC)
+  message(STATUS "  - dylib-sync")
+endif()
+if(IREE_HAL_DRIVER_VMVX)
+  message(STATUS "  - vmvx")
+endif()
+if(IREE_HAL_DRIVER_VMVX_SYNC)
+  message(STATUS "  - vmvx-sync")
+endif()
+if(IREE_HAL_DRIVER_VULKAN)
+  message(STATUS "  - vulkan")
+endif()
```

Comment:
Printing one driver/backend per line is pretty spammy IMO (I think I also received that feedback when editing this last, I pointed to https://dawn.googlesource.com/tint/+/refs/heads/main/CMakeLists.txt#103 as precedent in the ecosystem).

Could change to `VERBOSE` or `DEBUG`: https://cmake.org/cmake/help/latest/command/message.html#general-messages

Reply:
I think it is pretty important to print (LLVM prints one per line). Given the possibility for silent failures and the need to post-mortem debug these things, this is some of the most important things to be printing.

Reply:
Found the earlier discussion: https://github.com/google/iree/pull/7936#discussion_r774207457

Reply:
Prior to this patch, I've manually edited this to fill status multiple times to debug things going on. It is particularly annoying when getting a log from elsewhere and not knowing how it was compiled. I'll reduce some other spammy things but this is actually important for supportability in the field.